### PR TITLE
Fixed: do not require a Content-Length header

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
The HTTP service in multiple places expected a Content-Length header or it would refuse to parse the body.

This breaks streaming (see [test failures](https://github.com/dapr/dapr/actions/runs/5446419269/jobs/9907127023)), since in case the data is streamed it's not possible to know the content's byte length in advance. The Content-Length header is not required per the HTTP specs, so we should not enforce its presence.

This fixes the issue by not relying on the Content-Length header's presence.

### CODE REVIEW

File: service/http/topic.go [hunk 1]
```diff
@@ -67,6 +67,52 @@ type topicEventJSON struct {
 	PubsubName string `json:"pubsubname"`
 }
 
+func (in topicEventJSON) GetData() (data any, rawData []byte) {
```
Comment (ItalyPaleAle) [inline service/http/topic.go:70]: Why this change, where I split this out in a separate function, you may ask?

Well, ask the linter and its complaining about "cyclomatic complexity"...

Comment (artursouza) [reply service/http/topic.go:70]: Can it be a private method instead?

Comment (ItalyPaleAle) [reply service/http/topic.go]: Sure, done

**Review by artursouza — APPROVED** [review]
